src/fondant/pipeline.py

```diff
@@ -49,44 +49,41 @@ class ComponentOp:
              https://kubeflow-pipelines.readthedocs.io/en/1.8.13/source/kfp.dsl.html
     """
 
+    COMPONENT_SPEC_NAME = "fondant_component.yaml"
+
     def __init__(
         self,
-        component_spec_path: t.Union[str, Path],
+        component_dir: t.Union[str, Path],
         *,
-        local_component: bool = True,
         arguments: t.Optional[t.Dict[str, t.Any]] = None,
         number_of_gpus: t.Optional[int] = None,
         node_pool_name: t.Optional[str] = None,
         p_volumes: t.Optional[t.Dict[str, k8s_client.V1Volume]] = None,
         ephemeral_storage_size: t.Optional[str] = None,
-    ):
-        self.component_spec_path = component_spec_path
-        self.local_component = local_component
+    ) -> None:
+        self.component_dir = Path(component_dir)
         self.arguments = arguments or {}
+
+        self.component_spec = ComponentSpec.from_file(
+            self.component_dir / self.COMPONENT_SPEC_NAME,
+        )
+        self.arguments.setdefault("component_spec", self.component_spec.specification)
+
         self.number_of_gpus = number_of_gpus
         self.node_pool_name = node_pool_name
         self.p_volumes = p_volumes
         self.ephemeral_storage_size = ephemeral_storage_size
 
-        self.extend_arguments()
-
     @property
-    def component_spec(self) -> ComponentSpec:
-        return ComponentSpec.from_file(self.component_spec_path)
-
-    def extend_arguments(self):
-        """Add the component specification to the arguments if not already present."""
-        if not self.arguments.get("component_spec"):
-            self.arguments["component_spec"] = json.dumps(
-                self.component_spec.specification,
-            )
+    def dockerfile_path(self) -> t.Optional[Path]:
+        path = self.component_dir / "Dockerfile"
```

Comment:
`Dockerfile` is the default and should be used but I have seen people use `dockerfile` before (and it also works) so maybe we should also check for that ?

Reply:
I don't think we have to, since we can expect these components to be implemented for `fondant` specifically. So we can impose our own expectations. I don't think that's a bad thing to do here, since we adhere to the standard and keep things simple.